feat: add PREPEND_LOCALE_PATHS configuration setting

keithgg opened the pull request:
## Description


edx-platform currently supports COMPREHENSIVE_THEME_LOCALE_PATHS setting, which appends paths to the end of LOCALE_PATHS, but there's currently no way to add additional paths to the start of the list to override the default translations.

In this PR, we added the PREPEND_LOCALE_PATHS setting to achieve this.

- Affects the Operator

## Supporting information

- PR for the `configuration` repo. https://github.com/openedx/configuration/pull/6677
- [SE-5299](https://tasks.opencraft.com/browse/SE-5299)
- https://pr29851.sandbox.opencraft.hosting/

## Testing instructions

On the docker devstack:
- `make dev.shell.lms`
- Edit the file `/edx/etc/lms.yml`. Add a `PREPEND_LOCALE_PATHS` setting like
```yaml
PREPEND_LOCALE_PATHS: [/edx/src/]
```
- Copy a translation file into that folder.
```bash
cp -r /edx/app/edxapp/edx-platform/conf/locale/fr/ /edx/src/es
```
Note that this is copying French to Spanish folder. It makes seeing the differences easier.
- In your browser load the LMS, and change the `openedx-language-cookie`'s value to `es`.
- Restart your lms dev server.
- If done correctly you should get the french translations loaded (it sometimes takes 5 minutes to load).

## Deadline

"None"

## Other information

**Settings**
```yaml
configuration_version: keith/se_5299_prepend_local_paths
configuration_source_repo_url: https://github.com/open-craft/configuration
edx_platform_repository_url: https://github.com/open-craft/edx-platform.git
```

openedx-webhooks commented:
Thanks for the pull request, @keithgg! I've created [OSPR-6427](https://openedx.atlassian.net/browse/OSPR-6427) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

keithgg committed:
feat: add PREPEND_LOCALE_PATHS configuration setting

edx-platform supports COMPREHENSIVE_THEME_LOCALE_PATHS setting, which
appends paths to the end of LOCALE_PATHS,  but there's currently no
way to add additional paths to the start of the list.

We want to be able to prepend locale paths so that we can override
existing translations in edx-platform.

natabene commented:
@keithgg Thank you for your contribution. I will line this up for my review.

rgraber reviewed (approved):
LGTM

openedx-webhooks commented:
@keithgg 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/29851&entry.1484655318=https://github.com/keithgg&entry.752974735=2022-02-01+07:40&entry.1917517419=2022-02-09+18:30&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
